activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -8,6 +8,8 @@ class Module
   # Defines a class attribute and creates a class and instance reader methods.
   # The underlying class variable is set to +nil+, if it is not previously
   # defined.
+  # Note: Both methods will be public even if this method is called from
+  # inside a private or protected block.
```

Comment:
What would you name "the area that follows a `private` or `protected` declaration"? I'd be happy to change it to something else but couldn't figure out a better terminology for it. 
This may be expected (although still a bit surprising) for class/module methods, however these functions also create instance methods that are also public which is unexpected. I will try to re-word this to reduce the strength of the statement.